pkg/plugins/mellanox/mellanox_plugin.go

```diff
@@ -189,7 +196,13 @@ func (p *MellanoxPlugin) Apply() error {
 		return nil
 	}
 	log.Log.Info("mellanox plugin Apply()")
-	return p.helpers.MlxConfigFW(attributesToChange)
+	if err := p.helpers.MlxConfigFW(attributesToChange); err != nil {
+		return err
+	}
+	if vars.MlxPluginFwReset {
+		return p.helpers.MlxResetFW(pciAddressesToReset)
```

Comment:
return an error here can be problematic if we have multiple PCIAddress and one them them failed (for example the primary one)

Reply:
If I understand you correctly your idea would be to introduce a `errs` variable to append the errors and return it at the end of the function, right?